packages/frontend/src/components/DashboardFilter/ActiveFilters/index.tsx

```diff
@@ -52,13 +54,15 @@ const ActiveFilters: FC<ActiveFiltersProps> = ({
 
     return (
         <>
-            {dashboardFilters.dimensions
-                .filter((item) => !!fieldsWithSuggestions[item.target.fieldId])
-                .map((item, index) => (
```

Comment:
We were filtering out invalid dashboard filters and then relying on the `index` to update and remove the filters. 

eg: 

all filters = [ invalid_filter, valid_filter ]
valid filters = [valid_filter] 

So on update/remove it would use index `0`, affecting the invalid_filter instead of the valid filter.